yt/frontends/swift/fields.py

```diff
@@ -0,0 +1,27 @@
+from yt.frontends.sph.fields import SPHFieldInfo
+
+
+class SwiftFieldInfo(SPHFieldInfo):
+    def __init__(self, ds, field_list, slice_info=None):
+        self.known_particle_fields += (
+            (
+                "InternalEnergies",
+                ("code_specific_energy", ["specific_thermal_energy"], None),
+            ),
+            ("Densities", ("code_mass / code_length**3", ["density"], None)),
+            ("SmoothingLengths", ("code_length", ["smoothing_length"], None)),
+        )
+        super().__init__(ds, field_list, slice_info)
+
+    def setup_particle_fields(self, ptype, *args, **kwargs):
+        super().setup_particle_fields(ptype, *args, **kwargs)
+
+        if ptype in ("PartType0", "Gas"):
+            self.setup_gas_particle_fields(ptype)
+
+    def setup_gas_particle_fields(self, ptype):
+        self.alias((ptype, "temperature"), (ptype, "Temperatures"))
+        self.alias(("gas", "temperature"), (ptype, "Temperatures"))
```

Comment:
one more backward compatibility concern -- as written, wouldn't this would always alias `(ptype, "temperature")` to `(ptype, "Temperatures")` even for the older swift datasets (which will not exist)? I'm not positive this is an issue... but it looks like the existing tests only utilize the `("gas", "density")` field, so the test suite wouldn't catch any errors from this (and I'm on the wrong computer to test it manually right now...). 

I think adding a check for the existence of the field similar to what the base `setup_particle_fields` does would be the way to go: 

https://github.com/yt-project/yt/blob/13dc4fbcba564439c90e9bdd8db013f797dfc9a9/yt/fields/field_info_container.py#L106-L107

The position aliasing below should be safe.

Reply:
note: I'm back at a computer with test data, I'm not actually sure my comment above is an issue. will update here in a bit...

Reply:
oooook, this is **not** an issue, because the first couple lines of the `alias` function check for the existence of the field that you're trying to create the alias for.

Reply:
Glad to see this is not an issue. Aliasing like this felt less like a proper solution and more like a hack to get it to work.